t/macros_math/nondecimal_base.t

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env perl
+
+=head1 nondecimal_base
+
+Tests conversion of integers to non-decimal bases.
+
+=cut
+
+use Test2::V0 '!E', { E => 'EXISTS' };
+
+die "PG_ROOT not found in environment.\n" unless $ENV{PG_ROOT};
+do "$ENV{PG_ROOT}/t/build_PG_envir.pl";
+
+loadMacros('nondecimal_base.pl');
+
+subtest 'conversion from a non-decimal base to base 10' => sub {
+	is convertBase('101010', from => 2),  42,   'convert from base 2';
+	is convertBase('44011',  from => 5),  3006, 'convert from base 5';
+	is convertBase('5073',   from => 8),  2619, 'convert from base 8';
+	is convertBase('98A',    from => 12), 1402, 'convert from base 12';
+	is convertBase('98T',    from => 12, digits => [ '0' .. '9', 'T', 'E' ]), 1402,
+		'convert from base 12 with non-standard digits';
+	is convertBase('9FE8', from => 16), 40936, 'convert from base 16';
+};
+
+subtest 'Convert from decimal to non-decimal bases' => sub {
+	is convertBase(12, to => 2), '1100',   'convert to base 2';
+	is convertBase(47, to => 2), '101111', 'convert to base 2';
+
+	is convertBase(98,  to => 5), '343',   'convert to base 5';
+	is convertBase(761, to => 5), '11021', 'convert to base 5';
+
+	is convertBase(519,  to => 8), '1007', 'convert to base 8';
+	is convertBase(2023, to => 8), '3747', 'convert to base 8';
+
+	is convertBase(853,  to => 12), '5B1',  'convert to base 12';
+	is convertBase(2023, to => 12), '1207', 'convert to base 12';
+	is convertBase(1678, to => 12, digits => [ '0' .. '9', 'T', 'E' ]), 'E7T',
+		'convert to base 12 using non-standard digits';
+
+	is convertBase(5752,  to => 16), '1678', 'convert to base 16';
+	is convertBase(41446, to => 16), 'A1E6', 'convert to base 16';
+};
+
```

Comment:
Should there be tests that include both `to` and `from`?  The code allows that.  Of course, they must both use the same `digits`, so I'm not sure if that is a limitation that needs addressing or not.

Reply:
I thought of this and for some unknown reason didn't add such a test.  I will.